File: Backend/schemas/player_s.py
from marshmallow import Schema, fields, validate

class PlayerSchema(Schema):
    id = fields.UUID(dump_only=True)
    name = fields.String(required=True)
    last_name = fields.String(required=True)
    flat_no = fields.String(required=True)
    age = fields.Integer(required=True, validate=validate.Range(min=1, max=100))
    mobile_number = fields.String(required=True, validate=validate.Length(equal=10))
    img_url = fields.String(allow_none=True)
    bowler_type = fields.String(required=True)
    batter_type = fields.String(required=True)
    category = fields.String(required=True, validate=validate.OneOf(['gold', 'silver', 'bronze']))
    status = fields.String(validate=validate.OneOf(['sold', 'unsold','available']))
    tournament_id = fields.UUID(required=True)

File: Frontend/src/api/players.ts
import api, { privateApi } from './axios';

export type Player = {
  id: string;
  name: string;
  lastName: string;
  flatNo: string;
  age: number;
  mobileNumber: string;
  imgUrl: string | null;
  bowlerType: 'left' | 'right' | null;
  batterType: 'left' | 'right' | null;
  category: 'gold' | 'silver' | 'bronze' | null;
  tournamentId: string;
  status:string;
  isOwner: boolean;
};

export const fetchPlayers = async (): Promise<Player[]> => {
  const response = await privateApi.get('players'); // Assumes Flask route is /players
  return response.data.map((player: any) => ({
    id: player.id,
    name: player.name,
    lastName: player.last_name,
    flatNo: player.flat_no,
    age: player.age,
    mobileNumber: player.mobile_number,
    imgUrl: player.img_url,
    bowlerType: player.bowler_type,
    batterType: player.batter_type,
    category: player.category,
    tournamentId: player.tournament_id,
    status:player.status,
    isOwner: player.is_owner,
  }));
};
